### PR TITLE
Pass minimum target compile options through `raft::raft`

### PR DESCRIPTION
**Pull request by cjnolet**
cc @robertmaynard @MatthiasKohl

### CODE REVIEW

File: cpp/CMakeLists.txt [hunk 1]
```diff
@@ -223,6 +223,8 @@ target_link_libraries(
 )
 
 target_compile_features(raft INTERFACE cxx_std_17 $<BUILD_INTERFACE:cuda_std_17>)
+target_compile_options(raft PUBLIC $<$<COMPILE_LANGUAGE:CUDA>:--expt-extended-lambda --expt-relaxed-constexpr>)
```
Comment (robertmaynard) [inline cpp/CMakeLists.txt:226]:
```suggestion
target_compile_options(raft PUBLIC $<$<COMPILE_LANGUAGE_AND_ID:CUDA,NVIDIA>:--expt-extended-lambda --expt-relaxed-constexpr>)
```

**Review by robertmaynard — CHANGES_REQUESTED** [review]

File: cpp/CMakeLists.txt (hunk 1 above)
Comment (robertmaynard) [reply cpp/CMakeLists.txt:226]: Since this is public we should be extra safe and only apply this to `nvcc` and not clang ( for the edgiest of edge cases )

Comment (cjnolet) [reply cpp/CMakeLists.txt]: Thanks Robert! Just committed that change

**Comment by cjnolet** [comment]
@robertmaynard it looks like we are getting an error in CI related to the changes. Is there something else I need to set in order to make that work? 

```
  Expression did not evaluate to a known generator expression

```

File: cpp/CMakeLists.txt [hunk 2]
```diff
@@ -223,6 +223,10 @@ target_link_libraries(
 )
 
 target_compile_features(raft INTERFACE cxx_std_17 $<BUILD_INTERFACE:cuda_std_17>)
+target_compile_options(
+  raft INTERFACE $<$<COMPILE_LANGUAGE_AND_ID:CUDA,NVIDIA>:--expt-extended-lambda
```
Comment (robertmaynard) [inline cpp/CMakeLists.txt:227]:
```suggestion
  raft INTERFACE $<$<COMPILE_LANG_AND_ID:CUDA,NVIDIA>:--expt-extended-lambda
```
Here is the correct generator expression.

**Review by robertmaynard — CHANGES_REQUESTED** [review]

**Pull request merged**